system-test/spanner.ts

```diff
@@ -527,6 +557,69 @@ describe('Spanner', () => {
       });
     });
 
+    describe('numerics', () => {
+      before(async function () {
+        if (IS_EMULATOR_ENABLED) {
+          this.skip();
+        }
+      });
+
+      it('should write numeric values', done => {
+        const value = Spanner.numeric('3.141592653');
+
+        insert({NumericValue: value}, (err, row) => {
+          assert.ifError(err);
+          assert.deepStrictEqual(row.toJSON().NumericValue, value);
+          done();
+        });
+      });
+
+      it('should write null numeric values', done => {
+        insert({NumericValue: null}, (err, row) => {
+          assert.ifError(err);
+          assert.strictEqual(row.toJSON().NumericValue, null);
+          done();
+        });
+      });
+
+      it('should throw for out of bounds values', done => {
+        insert({NumericValue: Spanner.numeric('3.1415926535')}, err => {
+          assert.strictEqual(err.code, grpc.status.FAILED_PRECONDITION);
+          done();
+        });
+      });
+
+      it('should write empty numeric array values', done => {
+        insert({NumericArray: []}, (err, row) => {
+          assert.ifError(err);
+          assert.deepStrictEqual(row.toJSON().NumericArray, []);
+          done();
+        });
+      });
+
+      it('should write null numeric array values', done => {
+        insert({NumericArray: [null]}, (err, row) => {
+          assert.ifError(err);
+          assert.deepStrictEqual(row.toJSON().NumericArray, [null]);
+          done();
+        });
+      });
+
+      it('should write numeric array values', done => {
+        const values = [
+          Spanner.numeric('-99999999999999999999999999999.999999999'),
```

Comment:
nit: could we also include a null value in the array to ensure that those are also encoded and decoded correctly?

Reply:
I'm happy to do so, but just curious if you think the 'should write null numeric array values' test just above this one is not sufficient. For some reason all the other 'should write <type> array values' tests are like this where they only include non-null values in the test.

Reply:
Sorry, you're right. I misread the above test case as 'write a null array', and not as 'write an array containing one null element'.